its/plugin/projects/suppress-warnings/kotlin/file2.kt

```diff
@@ -35,3 +35,79 @@ fun complexity() {
         }
     }
 }
+
+fun matchingValue(value: Int): String {
+    @Suppress("kotlin:S1479")
+    return when(value) {
+        0 -> "0"
+        1 -> "0"
+        2 -> "0"
+        3 -> "0"
+        4 -> "0"
+        5 -> "0"
+        6 -> "0"
+        7 -> "0"
+        8 -> "0"
+        9 -> "0"
+        10 -> "0"
+        11 -> "0"
+        12 -> "0"
+        13 -> "0"
+        14 -> "0"
+        15 -> "0"
+        16 -> "0"
+        17 -> "0"
+        18 -> "0"
+        19 -> "0"
+        20 -> "0"
+        21 -> "0"
+        22 -> "0"
+        23 -> "0"
+        24 -> "0"
+        25 -> "0"
+        26 -> "0"
+        27 -> "0"
+        28 -> "0"
+        29 -> "0"
+        30 -> "0"
+        else -> "Unknown"
+    }
+}
+
+@Suppress("kotlin:S1479")
+fun matchingValue(value: Int): String {
```

Comment:
Are we sure this parses correctly (the signatures of the two methods seem to be identical)? 
Can we add 1 case where we don't suppress to this file and check we do trigger the rule at least once?